### ISSUE TITLE
MSCloudLoginAssistant: Could not load file or assembly Newtonsoft.Json, Version=13.0.0.0

### ISSUE THREAD

**Issue by Darian-FRB** [microsoft/Microsoft365DSC#2521]
#### Details of the scenario you tried and the problem that is occurring
I am required to operate on a management server running Windows Server 2019 Standard (Version	10.0.17763 Build 17763). PowerShell 7.1.7 is installed along with Microsoft365DSC 1.22.1109.1 and its dependencies (Update-M365DSCDependencies).

When attempting to export a DSC configuration I encounter the following error:

#### Verbose logs showing the problem
Connect-MgGraph: C:\Program Files\PowerShell\Modules\MSCloudLoginAssistant\1.0.97\Workloads\MicrosoftGraph.psm1:102
Line |
 102 |  …             Connect-MgGraph -AccessToken $Global:MSCloudLoginConnecti …
        |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        | Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral,
        | PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified.

#### Suggested solution to the issue
`[System.AppDomain]::CurrentDomain.GetAssemblies() | select FullName, Location` yields Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed

Newtonsoft.Json version 13.0.1 is mandatory due to a security vulnerability, https://github.com/advisories/GHSA-5crp-9r3c-p9vr.

In situations like this, a general solution is to upgrade to PowerShell 4.2, https://github.com/PowerShell/PowerShell/pull/15084.

However, Microsoft365DSC is only supported on PowerShell 4.1.

I am not knowledgeable enough with PowerShell to know where to proceed with troubleshooting. If I am on the right course, I am unsure why the 13.0.1 version of Newtonsoft.Json included with Microsoft.Graph.Authentication is not being used. I have confirmed it is in the $Env:PSModulePath.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)

```
# Generated by Microsoft365DSC from https://export.microsoft365dsc.com on 11/9/2022, 1:55:08 PM
# Visit https://microsoft365dsc.com for more information

# Getting application information for Application + Secret authentication
$ApplicationId = Read-Host -Prompt 'Application Id'
$ApplicationSecret = Read-Host -Prompt 'Application Secret'
$TenantId = Read-Host -Prompt 'Tenant Id'

# Exporting resources using application
Export-M365DSCConfiguration -Components @("AADAdministrativeUnit", "AADApplication", "AADAuthorizationPolicy", "AADConditionalAccessPolicy", "AADEntitlementManagementAccessPackageCatalog", "AADEntitlementManagementAccessPackageCatalogResource", "AADGroup", "AADGroupLifecyclePolicy", "AADGroupsNamingPolicy", "AADGroupsSettings", "AADNamedLocationPolicy", "AADRoleDefinition", "AADRoleSetting", "AADSecurityDefaults", "AADServicePrincipal", "AADTenantDetails", "AADTokenLifetimePolicy", "AADUser") -ApplicationId $ApplicationId -ApplicationSecret $ApplicationSecret -TenantId $TenantId
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2019 Standard
OsOperatingSystemSKU : StandardServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.1109.1

**Comment by NikCharlebois** [comment]
This is normally due to different versions of the Microsoft.Graph.* modules installed. In most cases you will have conflicting versions installed under:
C:\Users\<user>\OneDrive - Microsoft\Documents\PowerShell\Modules
and
C:\Program FIles\WindowsPowerShell\Modules

Our recommendation is to remove all modules under the first path above (the OneDrive one) and make sure to only keep those in the Program Files

**Comment by Darian-FRB** [comment]
I had checked other module installation paths, there is only a single instance of Microsoft.Graph.* in C:\Program Files\PowerShell\Modules.

There are no other instances of Microsoft.Graph.* in the other module paths:
C:\Program Files\PowerShell\7\Modules
C:\Program Files\WindowsPowerShell\Modules
C:\Users\ [user] \OneDrive - [tenant] \Documents\PowerShell\Modules
C:\Users\ [user] \Documents\PowerShell\Module
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
C:\Program Files\SharePoint Online Management Shell\
D:\Program Files\Microsoft Monitoring Agent\Agent\PowerShell\

I have also confirmed the Update-M365DSCDependencies version of the Microsoft.Graph.* modules are loaded in the session.

**Comment by NikCharlebois** [comment]
When you run:

Get-Module Microsoft.Graph.* -ListAvailable

Are they all under the C:\Program FIiles\WindowsPowerShell folder?

**Comment by Darian-FRB** [comment]
> When you run:
> 
> Get-Module Microsoft.Graph.* -ListAvailable
> 
> Are they all under the C:\Program FIiles\WindowsPowerShell folder?

No. They are in C:\Program Files\PowerShell\Modules. This is where Update-M365DSCDependencies and Install-Module places them.

**Comment by andikrueger** [comment]
Could you try to run Uninstall-M365DSCOutdatedDependencies, Just to make sure there.

**Comment by Darian-FRB** [comment]
I have run Uninstall-M365DSCOutdatedDependencies. No change :(

**Comment by andikrueger** [comment]
Could you run a test using PS5.1?

**Comment by andikrueger** [comment]
Could you re-test with PS7.1 now :)

**Comment by NikCharlebois** [comment]
We will need to investigate further. Looks like the recent updates to the Microsoft.Graph PowerShell SDK are no longer working with PowerShell 7.1 @peombwa 

In the meantime, you can update to PowerShell 7.3 to do your export, which works for EXport-M365DSCConfiguration

**Comment by Darian-FRB** [comment]
Oddly, Update-M365DSCDependencies (1.22.1123.1) under PS 7.3.0 did not install the required updates. Manually installing them resolved this issue. Your assistance is appreciated.

**Comment by Darian-FRB** [comment]
Also, https://github.com/microsoft/Microsoft365DSC/issues/2209 still applies.
